src/ember-app.js

```diff
@@ -210,6 +222,19 @@ class EmberApp {
       fastbootInfo: fastbootInfo
     });
 
+    let destroyAppInstanceTimer;
+    if (destroyAppInstanceInMs) {
+      // start a timer to destroy the appInstance forcefully in the given ms.
+      // This is a failure mechanism so that node process doesn't get wedged if the `visit` never completes.
+      destroyAppInstanceTimer = setTimeout(function() {
```

Comment:
Personally I'm not a fan of using `setTimeout` server-side since it's not deterministic. I understand the rationale here but let's think about it to see if there's a better way of implementing this.